examples/lighting-app/nrfconnect/README.md

```diff
@@ -1,289 +1,333 @@
+    -   [Bluetooth LE Rendezvous](#bluetooth-le-rendezvous)
+    -   [Thread provisioning](#thread-provisioning)
+-   [Requirements](#requirements)
+-   [Device UI](#device-ui)
+-   [Setting up the environment](#setting-up-the-environment)
+    -   [Using Docker container for setup](#using-docker-container-for-setup)
+    -   [Using native shell for setup](#using-native-shell-for-setup)
+-   [Building](#building)
+-   [Configuring the example](#configuring-the-example)
+-   [Flashing and debugging](#flashing-and-debugging)
+-   [Testing the example](#testing-the-example)
 
 <hr>
 
--   [CHIP nRF52840 Lighting Example Application](#chip-nrf52840-lighting-example-application)
-    -   [Introduction](#introduction)
-    -   [Device UI](#device-ui)
-    -   [Building](#building)
-        -   [Using Docker container](#using-docker-container)
-        -   [Using Native shell](#using-native-shell)
-        -   [Supported nRF Connect SDK versions](#supported-nrf-connect-sdk-versions)
-    -   [Configuring the example](#configuring-the-example)
-    -   [Flashing and debugging](#flashing-and-debugging)
-    -   [Accessing the command line](#accessing-the-command-line)
+<a name="overview"></a>
+
+## Overview
+
+This example is running on the nRF Connect platform, which is based
+on the [nRF Connect SDK](https://developer.nordicsemi.com/nRF_Connect_SDK/doc/latest/nrf/index.html)
+and [Zephyr RTOS](https://zephyrproject.org/). Visit CHIP's
+[nRF Connect Platform Overview](TODO:...) to read more information about
+platform structure and dependencies.
+
+The CHIP device that runs the lighting application is controlled by the CHIP controller
+device over the Thread protocol. By default, the CHIP device should have Thread disabled,
+and it should be paired with CHIP controller and get configuration from it.
+Some actions required before establishing full communication are described below.
+
+The example also comes with a test mode, which allows to start Thread with the default settings
+by pressing button manually. However, this mode does not guarantee that the device will be able
+to communicate with the CHIP controller and other devices.
+
+### Bluetooth LE Advertising
+
+After powering up the device for the first time, it will start advertising over
+Bluetooth LE to inform other devices about its presence. For security reasons,
+devices can start advertising automatically or wait for user input.
+For this example, advertising is started by pressing **Button 4**.
+
+### Bluetooth LE Rendezvous
+
+In CHIP, there is a commissioning procedure called Rendezvous, which is done over
+Bluetooth LE between a CHIP device and the CHIP controller,
+where the controller has the commissioner role.
+
+To start the Rendezvous, the controller must get the onboarding information from the CHIP device.
+The data payload is encoded within a QR code and typically presented on the device's display.
+For this example, however, it is shared using the NFC tag.
```

Comment:
Hmm, i think we should mention that both QR Code and NFC tags can be used. Those are two viable options. Also previous sentence suggests that QR Code is the only way.